pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -78,6 +79,9 @@ let
                         then "package-db"
                         else "package-conf";
 
+  # the target dir for haddock documentation
+  docdir = docoutput: docoutput + "/share/doc";
```

Comment:
This utility function doesn't strike me as particular useful. It's used only two times and provides no significant simplification of the code, IMHO.